src/api/webhook.py

```
@@ -0,0 +1,89 @@
+    try:
+        # Process the webhook
+        events = handler.parse(body.decode('utf-8'), x_line_signature)
+        
+        for event in events:
+            if isinstance(event, MessageEvent) and isinstance(event.message, TextMessage):
+                await handle_text_message(event)
+        
+        return {"status": "success"}
+        
+    except InvalidSignatureError:
+        raise HTTPException(status_code=400, detail="Invalid signature")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)